lib/dslash_quda.cu

```diff
@@ -10,6 +10,9 @@
 #include <linalg.cuh>
 #include <dslash_policy.cuh>
 #include <instantiate.h>
+#ifdef NVSHMEM_COMMS
```

Comment:
Random thought I had here, should we put this in the `shmem_helper`/some other common relevant header so we don't have this in multiple places? (I imagine this will happen one way or another in `generic_kernels` anyway.)

Reply:
The `#ifdef NVSHMEM_COMMS` ?

Reply:
The `#include <cuda/atomic>`

Reply:
(and corresponding ifdef wrapper)

Reply:
No, we don't want to include the shmem_helper in device code without device-linking ()i.e. code that just needs the atomic) as it will warn about `extern` being treated as static (in the nvshmem headers). So I'd like to keep that separate for now. But worth looking into further cleanup and see how things evolve with atomic usage.

Reply:
#1118